sgt-back-end/index.js

```diff
@@ -0,0 +1,142 @@
+const express = require('express');
+const exJSON = express.json();
+const app = express();
+app.use(exJSON);
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+// returns all rows from the "grades" table. Client should receive an array of objects.
+// if there are no rows, returns an empty array
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+SELECT *
+FROM "grades"
+ORDER BY "gradeId" ASC;
+`;
+  db.query(sql)
+    .then(result => {
+      const gradesTable = result.rows;
+      res.status(200).json(gradesTable);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+// POST - inserts a new grade into "grades" table and return the created grade.
+// The client should receive and object, not an array
+app.post('/api/grades', (req, res, next) => {
+  // status 400 if client supplies an invalid grade, including a missing name, course or score.
+  if (!req.body || !req.body.name || !req.body.course || !req.body.score) {
+    const errObj = { error: 'missing all or some of name, course, score query parameters' };
+    return res.status(400).json(errObj);
+  } else if (req.body.score < 0 || req.body.score > 100 || isNaN(req.body.score)) { // Or score isnt an integer from 0 to 10
+    const errObj = { error: 'score value is not an integer from 0 to 100 or is NaN' };
+    return res.status(400).json(errObj);
```

Comment:
This won't stop non-integer-numbers in the range (ex: `4.52` is not an integer, but it will pass through).